templates/gce.tf.erb

```diff
@@ -1,12 +1,6 @@
-provider "google" {
-  region      = "<%= ENV['GCE_REGION'] %>"
-  project     = "govuk-publishing-dns"
-  credentials = "${file("<%= ENV['GCE_CREDENTIALS'] %>")}"
-}
-
 <% records['records'].each_with_index do | record, index | %>
 resource "google_dns_record_set" "<%= record['subdomain'].gsub('.', '_').gsub('@', 'AT') %>_<%= index %>" {
```

Comment:
If the order in the YAML file ever changes the terraform statefile will rename the resources as the index changes and then remove and re-add the records. I think.